### PR TITLE
Feature: Make lonely emojis bigger

### LINKED ISSUE #3712: New Feature: Make lonely emojis bigger in E.cash
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
Emoji with no text are sent

## Expected Result:
Emoji to be larger than they normally are

## Actual Result:
Emoji are small

## Platform:
All (I assume)
 
**Version Number:**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 
From earlier post, providing for more context. 
# Problem
Emojis by themselves in E.cash are small and sometimes hard to see.

# Why is this important?
Having bigger emojis is a nice design feature and improves the UX of our chat app. This also provides more parity with Slack. 

# Solution
If the entire content of a message is emojis, then increase the font size so the emojis appear bigger. Here is an example of what this looks like on Slack:

image
**Expensify/Expensify Issue URL:**
https://github.com/Expensify/Expensify/issues/160734#

[View all open jobs on Upwork](https://www.upwork.com/ab/jobs/search/?q=Expensify%20React%20Native&sort=recency&user_location_match=2)

### PR DESCRIPTION
**Pull request by rdjuric**
### Details

Increases the fontSize of the `ReportActionItemFragment` of type `COMMENT` when it's composed of a single emoji. To do that, we: 
1. Add a new style: `singleEmojiText`, with custom fontSize and lineHeight
2. Add a new const: a regex expression to match emojis `CONST.REGEX.EMOJIS`
3. Create `isSingleEmoji`, a function that validates if a string is composed of a single emoji
4. In the `ReportActionItemFragment` we check if the `COMMENT` passes our `isSingleEmoji` validation. If it passes, we use the `singleEmojiText` style instead of our default.

### Fixed Issues

Fixes #3712 

### Tests

1. Open any conversation.
2. Send a message composed of a single emoji.
3. Send a second message composed of multiple emojis (or emoji + text)
3. The lonely emoji on the first message is bigger.
### QA Steps

1. Open any conversation.
2. Send a message composed of a single emoji.
3. Send a second message composed of multiple emojis (or emoji + text)
3. The lonely emoji on the first message should be bigger.
### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

#### Web

https://user-images.githubusercontent.com/53711423/123180220-3d485980-d461-11eb-99da-ac9554b056c3.mov

#### Mobile Web

https://user-images.githubusercontent.com/53711423/123185158-a634cf00-d46b-11eb-9d4b-8da36ac9ac82.mp4

#### Desktop

https://user-images.githubusercontent.com/53711423/123180672-19394800-d462-11eb-8a07-0f8610958488.mov

#### iOS

https://user-images.githubusercontent.com/53711423/123185354-1cd1cc80-d46c-11eb-8169-1b9a2c231368.mp4

#### Android

https://user-images.githubusercontent.com/53711423/123186929-8dc6b380-d46f-11eb-909a-e46c706d3f65.mp4

### CODE REVIEW

**Comment by rdjuric** [comment]
Just a note: while #3730 isn't fixed, the only way to test this is by sending messages while offline.

File: src/CONST.js [hunk 1]
```diff
@@ -332,6 +321,19 @@ const CONST = {
...
...\udfff|\u263a|\u2639|\u2620|\u2763|\u2764|\u270b|\u270c|\u261d|\u270a|\u270d|\u26f7|\u26f9|\u2618|\u2615|\u26f0|\u26ea|\u26e9|\u26f2|\u26fa|\u2668|\u26fd|\u2693|\u26f5|\u26f4|\u2708|\u231b|\u23f3|\u231a|\u23f0|\u23f1|\u23f2|\u2600|\u2b50|\u2601|\u26c5|\u26c8|\u2602|\u2614|\u26f1|\u26a1|\u2744|\u2603|\u26c4|\u2604|\u2728|\u26bd|\u26be|\u26f3|\u26f8|\u2660|\u2665|\u2666|\u2663|\u265f|\u26d1|\u260e|\u2328|\u2709|\u270f|\u2712|\u2702|\u26cf|\u2692|\u2694|\u2699|\u2696|\u26d3|\u2697|\u26b0|\u26b1|\u267f|\u26a0|\u26d4|\u2622|\u2623|\u2b06|\u2197|\u27a1|\u2198|\u2b07|\u2199|\u2b05|\u2196|\u2195|\u2194|\u21a9|\u21aa|\u2934|\u2935|\u269b|\u2721|\u2638|\u262f|\u271d|\u2626|\u262a|\u262e|\u2648|\u2649|\u264a|\u264b|\u264c|\u264d|\u264e|\u264f|\u2650|\u2651|\u2652|\u2653|\u26ce|\u25b6|\u23e9|\u23ed|\u23ef|\u25c0|\u23ea|\u23ee|\u23eb|\u23ec|\u23f8|\u23f9|\u23fa|\u23cf|\u2640|\u2642|\u26a7|\u2716|\u2795|\u2796|\u2797|\u267e|\u203c|\u2049|\u2753|\u2754|\u2755|\u2757|\u3030|\u2695|\u267b|\u269c|\u2b55|\u2705|\u2611|\u2714|\u274c|\u274e|\u27b0|\u27bf|\u303d|\u2733|\u2734|\u2747|\u00a9|\u00ae|\u2122|\u2139|\u24c2|\u3297|\u3299|\u26ab|\u26aa|\u2b1b|\u2b1c|\u25fc|\u25fb|\u25fe|\u25fd|\u25aa|\u25ab)/g,
```
Comment (parasharrajat) [inline src/CONST.js:335]: I thought this is going to be a short one. :laughing:

Comment (rdjuric) [reply src/CONST.js:335]: 😄 There're more short versions, but they don't really work for our use case. 

The reason they're shorter is that they work with unicode ranges, but by doing this a composed emoji (a emoji that is made from a sequence of other emojis, like 👩‍👩‍👧‍👦 =  👨👩👧👦) will match with all it's parts. So

'👨👩👧👦'.match(regex) will give the same result as '👩‍👩‍👧‍👦'.match(regex) and we wouldn't be able to differentiate between a single (but composed emoji) and a emoji sequence.

**Review by robertjchen — CHANGES_REQUESTED** [review]

File: src/CONST.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (rdjuric) [reply src/CONST.js:335]: I think it's more performant to use the literal notation here, but let me know what you think

From the [Mozilla Docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp)

> The literal notation results in compilation of the regular expression when the expression is evaluated. Use literal notation when the regular expression will remain constant. For example, if you use literal notation to construct a regular expression used in a loop, the regular expression won't be recompiled on each iteration.
> 
> The constructor of the regular expression object—for example, new RegExp('ab+c')—results in runtime compilation of the regular expression. Use the constructor function when you know the regular expression pattern will be changing, or you don't know the pattern and obtain it from another source, such as user input

File: src/libs/ValidationUtils.js [hunk 2]
```diff
@@ -17,6 +17,23 @@ function isValidAddress(value) {
     return !CONST.REGEX.PO_BOX.test(value);
 }
 
+/**
+ * Validates that this string is composed of a single emoji
+ *
+ * @param {String} string
```
[the remark this answers is not in the document]
Comment (rdjuric) [reply src/libs/ValidationUtils.js:23]: Good point! Will change it to message.

File: src/CONST.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (parasharrajat) [reply src/CONST.js:335]: I believe there is a small regex for Emoji.

Comment (parasharrajat) [reply src/CONST.js:335]: Check for length. nad No of surrogate pairs.

[the remark this answers is not in the document]
Comment (rdjuric) [reply src/CONST.js:335]: There're plenty. I explained why they don't work on your comment, but I'll add it here too: 

> 😄 There're more short versions, but they don't really work for our use case.
> 
> The reason they're shorter is that they use unicode ranges, but by doing this a composed emoji (a emoji that is made from a sequence of other emojis, like 👩‍👩‍👧‍👦 = 👨👩👧👦) will match with all it's parts. So
> 
> '👨👩👧👦'.match(regex) will give the same result as '👩‍👩‍👧‍👦'.match(regex) and we wouldn't be able to differentiate between a single (but composed emoji) and a emoji sequence.

[the remark this answers is not in the document]
Comment (parasharrajat) [reply src/CONST.js:335]: Also, this is going to be a very expensive operation. On each render, seems like a kill.

[the remark this answers is not in the document]
Comment (rdjuric) [reply src/CONST.js:335]: `ReportActionItemFragment` is a PureComponent. The regex runs in single digits ms.

Tested on a string with all emojis: 3.552ms
On a string with thousands of characters before any emoji: 5.066ms

[the remark this answers is not in the document]
Comment (parasharrajat) [reply src/CONST.js:335]: Then it's fine.

**Review by robertjchen — APPROVED** [review]
Looks great, thank you!

**Pull request merged**